### DISCUSSION TITLE
Refrence themes and assets in a central repository

### DISCUSSION THREAD

**Discussion by ghost** [marp-team/marp#482] [Q&A] [answered]
Background:
I create presentations in marp for vscode for various subjects and have a folder structure set up by subject.
Certain assets (theme.css-files, fonts, images, logos ...) constitute my branding and will remain the same for all presentations.

Problem:
Each time I start a new presentation, I have to place a .vscode directory with a settings.json file and a theme.css-file into the working directory's root  before I can start creating. This leads to versioning issues in the theme.css-files and duplication and is overall a bit messy.
How can I reference a **central repository** for all the assets that constitute my branding?

**Comment by yhatt** [comment]
Marp team leaves the way to manage your themes and assets to users :)

But if your brand is established, it's worth trying to use remote asset URLs served by localhost or CDN assets instead of the reference to local files.

Markdown documents and theme CSS can access to assets over the remote URL. In addition, [Marp for VS Code can fetch the theme via HTTPS url](https://github.com/marp-team/marp-vscode#use-custom-theme-css-%EF%B8%8F), as like as [VS Code's Markdown preview](https://code.visualstudio.com/docs/languages/markdown#_using-your-own-css).

```json
{
  "markdown.marp.themes": [
    "https://example.com/foo/bar/custom-theme.css",
  ]
}
```

```css
/* @theme custom-theme */
section {
  background: url('https://example.com/foo/bar/images/foobar.webp') no-repeat center / cover;
}
```

```markdown
---
marp: true
theme: custom-theme
---

# Hello ![w:1em h:1em](https://example.com/foo/bar/icons/icon.svg)

**Reply by codingluke** [reply]
How about to reference the theme directly via the github raw url? So you don't have to publish it on an extra domain?

**Reply by ghost** [reply]
I need to reference not only the theme.css-file, but also logo-files and some standard imagery. I am new to Github. so I didn't think about Github as a solution. Are you suggesting that I create a dedicated public github account for that? No major privacy issues here since the branding assets are meant to be seen by the public. It might only be a bit embarrassing when real pros look at my heavy-handed amateur-attempt at creating a marp-theme.

**Reply by yhatt** [reply, marked as answer]
It's useful to manage themes and assets on GitHub, but it's not always necessary. You can use any web hosting service, or your local server `localhost` served by any way.

@codingluke The raw URL on GitHub can be recognized as far as using Marp for VS Code to load themes. But for assets, referencing by `raw.githubcontent.com` may fail due to invalid MIME type. (ref. [FAQ by _deprecated_ RawGit](https://github.com/rgrove/rawgit/blob/master/FAQ.md#why-is-this-necessary-cant-i-just-load-files-from-github-directly))
